Python/generated_cases.c.h

```diff
@@ -2760,14 +2763,13 @@
                 Py_DECREF(enter);
                 goto error;
             }
-            SET_TOP(exit);
             Py_DECREF(mgr);
             res = _PyObject_CallNoArgs(enter);
             Py_DECREF(enter);
-            if (res == NULL) {
-                goto error;
-            }
-            PUSH(res);
+            if (res == NULL) goto pop_1_error;
```

Comment:
It's leaking when __enter__ raises.

Reply:
It leaks exit. Before this PR it put exit in the stack before the 'goto error', but now it doesn't so it needs to decref exit.